### ISSUE TITLE
Pin Consul version like Terraform Version Constraint model

### ISSUE THREAD

**Issue by npearce** [hashicorp/terraform-provider-hcp#299]
### Description

Please implement the terraform version constraint syntax to support:
a) specifying a specific version: `version = 1.11.5`
b) specify a desired major version: `version = ~> 1.11.5`.  <- To allow 1.11.5 or higher, but not 1.12x. 

From what I can see, I can only specify a `min_consul_version`, but I cannot prevent a major version upgrade when, for example, the HCP platform changes the `recommended_version` from `1.11.5` to `1.12.x`.

```
[min_consul_version](https://registry.terraform.io/providers/hashicorp/hcp/latest/docs/resources/consul_cluster#min_consul_version) (String) The minimum Consul version of the cluster. If not specified, it is defaulted to the version that is currently recommended by HCP.
```

### New or Affected Resource(s)

* hcp_consul_cluster

### Potential Terraform Configuration

```hcl
resource "hcp_hvn" "example" {
  hvn_id         = "hvn"
  cloud_provider = "aws"
  region         = "us-west-2"
  cidr_block     = "172.25.16.0/20"
}

resource "hcp_consul_cluster" "example" {
  cluster_id = "consul-cluster"
  hvn_id     = hcp_hvn.example.hvn_id
  tier       = "development"
  version = " ~> 1.11.5"
}```

### References

We already to this in TF: https://www.terraform.io/language/expressions/version-constraints

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

**Comment by bcmdarroch** [comment]
Thanks for contributing this issue and apologies for the late response! The HCP Consul team is currently working on the cluster upgrade experience, so this is valuable feedback.

**Comment by loshz** [comment]
This has now been fixed in #406

**Comment by paraspraj** [comment]
Hey all, just a heads up, we've released v0.48.0 of the HCP Terraform Provider which includes #406 
[v0.48.0](https://github.com/hashicorp/terraform-provider-hcp/releases/tag/v0.48.0)
